.github/workflows/windows_intel_oneAPI.yml

```diff
@@ -0,0 +1,115 @@
+    #        (b) it does run with Act locally (`github` doesn't exist there)
+    if: >
+      needs.get_commit_message.outputs.message == 1
+      && (github.repository == 'scipy/scipy' || github.repository == '')
+    runs-on: windows-2022
+    defaults:
+      run:
+        shell: powershell
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
+        with:
+          submodules: recursive
+
+      - name: Setup conda
+        uses: s-weigand/setup-conda@v1
+        with:
+          update-conda: true
+          python-version: 3.11
+      - run: conda --version
+      - run: which python
+
+      - name: Install packages from conda
+        run: |
+          conda install -c conda-forge pkg-config meson=1.4.0 meson-python=0.16.0 ninja openblas libblas=*=*openblas
+
+      - name: pip-packages
+        run: |
+          pip install numpy==1.23.5 cython pybind11 pytest pytest-xdist pytest-timeout pytest-fail-slow pooch rich_click click doit pydevtool hypothesis
+
+      - name: cache install
+        id: cache-install
+        uses: actions/cache@v3
+        with:
+          path: |
+              C:\Program Files (x86)\Intel\oneAPI\compiler
+          key: install-${{ env.WINDOWS_HPC_URL }}-${{ env.WINDOWS_BASEKIT_URL }}-compiler
+      - name: Install oneAPI Base kit
+        if: steps.cache-install.outputs.cache-hit != 'true'
+        run: |
+          echo %WINDOWS_BASEKIT_URL%
+          tools/install_intel_oneAPI_windows.bat %WINDOWS_BASEKIT_URL%
+      - name: Install oneAPI HPC kit
+        if: steps.cache-install.outputs.cache-hit != 'true'
+        run: |
+          echo %WINDOWS_HPC_URL%
+          tools/install_intel_oneAPI_windows.bat %WINDOWS_HPC_URL%
+
+      # The following steps are necessary because calling `.bat` files for Intel oneAPI and
+      # MSVC doesn't update the environment on CI. Both `ifx` and `cl` aren't detected resulting
+      # in failure of build setup for SciPy. After trying all the approaches (
+      # like calling `.bat` file from another `.bat` file, removing conda, adding conda),
+      # the only option left was to manually set the PATH, INCLUDE, and LIB.
+      # The OS is set to windows-2022 and therefore MSVC should also be fixed to the version
+      # being used in this check.
+      - name: Set PATH
+        run: python tools\activate_intel_oneAPI_PATH.py
+
+      - name: Set LIB
+        run: python tools\activate_intel_oneAPI_LIB.py
+
+      - name: Set INCLUDE
+        run: python tools\activate_intel_oneAPI_INCLUDE.py
+
+      - name: set FC
+        run: echo "FC=ifx" | Out-File -FilePath $env:GITHUB_ENV -Append
+
+      # MSVC throws error with Pythran in SciPy.
+      # So we use -C-Duse-pythran=false to skip Pythran dependency.
+      # Reference - https://github.com/serge-sans-paille/pythran/issues/2215
+      - name: Build
+        run: python dev.py build -C-Duse-pythran=false -C--vsenv
+        shell: cmd
+
+      - name: Test scipy.datasets
```

Comment:
Can you add a comment above here referencing the issue with the crash in `_arpack`, and that's why tests for other submodules are not run?